src/lender/tranche/senior_tranche.sol

```diff
@@ -36,17 +36,21 @@ contract SeniorTranche is Tranche, Interest {
 
     AssessorLike  public assessor;
 
-    function debt() external returns(uint) {
-        drip();
-        return safeAdd(borrowed, interest);
-    }
-
     constructor(address token_, address currency_, address assessor_) Tranche(token_ ,currency_) public {
         ratePerSecond = ONE;
         lastUpdated = now;
         assessor = AssessorLike(assessor_);
     }
 
+    function updatedDebt() external returns(uint) {
+        drip();
+        return safeAdd(borrowed, interest);
+    }
+
+    function debt() external view returns(uint) {
+        return safeAdd(borrowed, _calcInterest());
+    }
```

Comment:
biggest change.

all other contracts should use `updatedDebt` only the assessor uses `debt` as view. (and might call `drip` before)